test/e2e/network/service.go

```diff
@@ -1102,9 +1070,6 @@ var _ = SIGDescribe("Services", func() {
 
 	ginkgo.It("should work after restarting kube-proxy [Disruptive]", func() {
 		// TODO: use the ServiceTestJig here
-		e2eskipper.SkipUnlessProviderIs("gce", "gke")
-		e2eskipper.SkipUnlessSSHKeyPresent()
```

Comment:
These checks are a poor proxy for asking "is kube-proxy running in this very specific manner and are we capable of restarting it?"

I might be in favor of abstracting / being more specific about what we're checking for e.g. `skipUnlessClusterComponentRunsAsPodsAndE2ETestClientIsAuthorizedToDeleteThem(component)` or `skipUnlessClusterProviderSupportsComponentRestart(component)` etc.

But I don't think I'm in favor of dropping the up-front check

Reply:
From the previous version,  the case alreay treat kube-proxy as pod. 

https://github.com/kubernetes/kubernetes/blob/fa15799f7409b8644221c2b43a494c74aac0c5ee/test/e2e/network/service.go#L712

if kube-proxy running as process, no one will restart it. And case can't run.Here are the same. 

Now if no find component pod in function `getComponentPod`, it wil return  error and the case will end with error.
Of cause,  in the front of case,  checking whether component is deploy as pod will be more nice. I will give it a try :)

Reply:
> if kube-proxy running as process, no one will restart it. And case can't run.Here are the same.

init(1) ?
I would think anyone running kube-proxy as a process has it under something like systemd.

Reply:
yeah, if kube-proxy run as process, systemd maybe manage it.